aspnetcore/grpc/client.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Call gRPC services with the .NET client
+author: jamesnk
+description: Learn how to call gRPC services with the .NET gRPC client.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: jamesnk
+ms.date: 08/06/2019
+uid: grpc/client
+---
+# Call gRPC services with the .NET client
+
+A .NET gRPC client library is available in the [Grpc.Net.Client](https://www.nuget.org/packages/Grpc.Net.Client) NuGet package. This document explains how to:
+
+* Configure a gRPC client to call gRPC services.
+* Make gRPC calls to unary, server streaming, client streaming, and bi-directional streaming methods.
+
+## Configure gRPC client
+
+gRPC clients are concrete client types that are [generated from *\*.proto* files](xref:grpc/basics#generated-c-assets). The concrete gRPC client has methods that translate to the gRPC service in the *\*.proto* file.
+
+A gRPC client is created from a channel. Start by using `GrpcChannel.ForAddress` to create a channel, and then use the channel to create a gRPC client:
+
+```csharp
+var channel = GrpcChannel.ForAddress("https://localhost:5001");
+var client = new Greet.GreeterClient(channel);
+```
+
+A channel represents a long-lived connection to a gRPC service. When a channel is created it is configured with options related to calling a service. For example, the `HttpClient` used to make calls, the maximum send and receive message size, and logging are options on `GrpcChannelOptions` that can be passed to `GrpcChannel.ForAddress`. For a complete list of options, see [client configuration options](xref:grpc/configuration#configure-client-options).
+
+Creating a channel can be an expensive operation and reusing a channel for gRPC calls offers performance benefits. Multiple gRPC clients can be created from a channel, including different types of clients. Concrete gRPC client types are lightweight objects and can be created when needed.
```

Comment:
Should we add a note here that you can bump against the maximum concurrent streams per connection limit based on how you've configured your server?

Reply:
I want to investigate that more before writing guidance.